spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/util/Util.java

```diff
@@ -27,7 +25,20 @@ private Util() {
 	}
 
 	public static String extractDatabasePath(String parent) {
-		return parent.substring(0, StringUtils.ordinalIndexOf(parent, "/", 4));
+		//the parent looks like this: projects/{project_id}/databases/{database_id}/...
+		//and the database path is the first 4 segments, separated by /
+		StringBuilder sb = new StringBuilder();
+		int count = 0;
+		for (int i = 0; i < parent.length(); i++) {
+			char c = parent.charAt(i);
+			if (c == '/') {
+				count++;
+			}
+			if (count == 4) {
+				break;
+			}
+			sb.append(c);
```

Comment:
Since you're appending each character at a time anyway in a StringBuilder, might as well just do `String.split("/")` and then do return `arr[0] + "/" + arr[1] + ... + arr[3]`; would be more readable (only 2-3 lines total).

Reply:
You can even make it more efficient using the limit arg in String.split: https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int)

Reply:
I was trying to avoid unnecessary computation, since split allocates an array and then we need to combine the items. Also, split is a regex based method so the computational complexity is higher.

Reply:
I see, makes sense.